### ISSUE TITLE
How to apply this loss function to s2anet to speed up convergence？

### ISSUE THREAD

**Issue by xc-chengdu** [ming71/RIDet#5]
Hello, thank you very much for your contribution, we are very interested in your research work. But we try to transfer it to s2anet training own data, we notice loss_odm_bbox:nan. Please let me know what else needs to be done to apply it to s2anet, we would love to master this approach and look forward to hearing from you.

**Comment by ming71** [comment]
Thank you for your attention to my work.  Does the nan problem occur when you use QBB representation or OBB representation?
I've encountered the issue in OBB form when there are 0 or quite small values in denominators.

**Comment by xc-chengdu** [comment]
irst of all, I am also using the OBB representation, but not directly using the open source algorithm project you contributed. In fact, I just added the loss function you designed to s2anet, however, when training my own data, odm_loss keeps outputting nan. How did you implement this loss function embedded in s2anet? I would like to learn this approach rather than just going with the algorithm you contributed. Looking forward to your answer, good luck.

Translated with www.DeepL.com/Translator (free version)

**Comment by xc-chengdu** [comment]
First of all, I am also using the OBB representation, but not directly using the open source algorithm project you contributed. In fact, I just added the loss function you designed to s2anet, however, when training my own data, odm_loss keeps outputting nan. How did you implement this loss function embedded in s2anet? I would like to learn this approach rather than just going with the algorithm you contributed. Looking forward to your answer, good luck.

**Comment by ming71** [comment]
You can refer to my implementation in this repository. 
To solve the nan problem on your dataset, you can further set breakpoints (pdb or ipdb) at the loss calculation to check if there are wrong labels or predictions. Keep me informed if find anything wrong.

**Comment by xc-chengdu** [comment]
I followed your suggestion and carefully compared the implementation in your codebase and found that the loss function is only called in the configuration file and little has been changed in s2anethead.py. My data representation format also follows the data format in s2anet, and when I train my small amount of data, the code works fine, but loss_odm_bbox is always nan from the beginning to the end. I confirm that there is nothing wrong with my data, as it can get a good representation in s2anet.

**Comment by ming71** [comment]
Mmdetection decomposes the detection pipeline into independent modules, and RIL only modifies the loss function, so there is no modification to the s2anet_head. Just check the loss [here](https://github.com/ming71/RIDet/blob/master/mmdet/models/losses/hungarian_loss.py).
